woocommerce/woocommerce-ios
# [Manual Taxes M2] Fetch Tax Rates

Closes: #10558 


## Description


(Sorry for the big PR, most of the lines come from adding models and testing files that are straightforward to review).

With this PR we fetch the Tax Rates from the backend, creating all the necessary classes to store and use them. We also refactor the former TaxClass classes (`TaxClassStore`, `TaxClassAction`, `TaxClassRemote`) to include the `TaxRate` logic implemented here.

## Testing instructions

As this is not used yet, there's no way to see this in the UI, but can place this code in any place (e.g. at the end of the [EditableOrderViewModel](https://github.com/woocommerce/woocommerce-ios/blob/4fd2dc81586992549801a96f53b269cd9924aa29/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/EditableOrderViewModel.swift#L376) `init`):

```
stores.dispatch(TaxAction.retrieveTaxRates(siteID: siteID, pageNumber: 1, pageSize: 25) { result in
            if case let .success(rates) = result {
                debugPrint("tax rates", rates)
            }
        })
```



## Screenshots

N/A


---
